src/components/tl/ucp/allreduce/allreduce_knomial.c

```diff
@@ -199,8 +199,9 @@ ucc_status_t ucc_tl_ucp_allreduce_knomial_init_common(ucc_tl_ucp_task_t *task)
     task->super.post     = ucc_tl_ucp_allreduce_knomial_start;
     task->super.progress = ucc_tl_ucp_allreduce_knomial_progress;
     task->super.finalize = ucc_tl_ucp_allreduce_knomial_finalize;
-    status = ucc_mc_alloc(&task->allreduce_kn.scratch, (radix - 1) * data_size,
```

Comment:
We don't need another variable "mc_header" in the allreduce_kn struct. You can just make "scratch" to be "mc_header_t" type. And then use scratch->addr in the code

Reply:
To be aligned with scratch in reduce_scatter_kn. 
In reduce_scatter_knomial.c there are many more usages of scratch which would call for many changes to scratch->addr in code. I thought one change in code is cleaner than many.

Reply:
But this way you add additional variable to the task structure thus making it larger in memory. It is better to make more line changes than increase the size of the data structure in the critical path just because too many lines to change.